### PR TITLE
[OPS-2112] environment unification

### PR DESCRIPTION
**Pull request by obscure-shadow**
Ticket: [OPS-2112](https://juiceanalytics.atlassian.net/browse/OPS-2112)
Type: Improvement

#### This PR introduces the following changes

- Remove environment folder structure
- Re-structured to be able to pull any tag from command line
- Made interactive environment selection work with new structure
- Added check for top level fruition symlink
- I might have butchered these tests so if they should be a different way I'm open to advice on improvements

## Documentation

Jb start is now run from the top level of devlandia directory, This can be called with all the usual suspects of environment declarations. It can also be used to declare a specific tag that was not previously easily available, eg `jb start release-3.63.4-py3` will pull the corresponding ecr tag if it exists

## Checklist

- [x] Add new feature to the usage documentation
- [x] Add tests/ fix tests

### CODE REVIEW

**Comment by obscure-shadow** [comment]
just noticed the ssh stuff is broken i'm working on a fix now

File: jbcli/jbcli/cli/jb.py [hunk 1]
```diff
@@ -546,8 +472,8 @@ def get_environment_interactively(env):
 def stop(ctx, clean, env):
     """Stop a running juicebox in this environment
     """
-    env = get_environment_interactively(env)
-    os.chdir("./environments/{}".format(env))
+    # env = get_environment_interactively(env)
+    # os.chdir("./environments/{}".format(env))
```
Comment (chrisgemignani) [inline jbcli/jbcli/cli/jb.py:476]: You can just delete these.

File: jbcli/jbcli/tests/test_cli.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 from ..cli.jb import DEVLANDIA_DIR, cli
 
 
-CORE_DIR = os.path.join(DEVLANDIA_DIR, 'environments/core')
+CORE_DIR = DEVLANDIA_DIR
```
Comment (chrisgemignani) [inline jbcli/jbcli/tests/test_cli.py:15]: Let's change all the references to CORE_DIR to reference DEVLANDIA_DIR and get rid of CORE_DIR

**Review by chrisgemignani — CHANGES_REQUESTED** [review]
A few small review comments. I'll test locally some more.

File: .env [hunk 3]
```diff
@@ -0,0 +1,4 @@
+DEVLANDIA_PORT=8000
```
Comment (radix) [inline .env:1]: I don't think you intended to include this file?

File: jbcli/jbcli/cli/jb.py [hunk 4]
```diff
@@ -439,39 +365,64 @@ def cleanup():
...
         ctx.invoke(upgrade)
 
-    stash.put('current_env', env)
-    os.chdir("./environments/{}".format(env))
+    stash.put('current_env', tag)
+    env_dot = open(".env", "w")
+    env_dot.write(f"DEVLANDIA_PORT=8000\nTAG={tag}\nFRUITION={core_path}\nFILE={core_end}")
+    env_dot.close()
+
     environ = populate_env_with_secrets()
```
Comment (radix) [inline jbcli/jbcli/cli/jb.py:413]: I think writing an `.env` file is an unnecessarily indirect way of setting these environment variables. We can just pass them in the environment when starting docker-compose.

Is there somewhere else this .env file is being relied on?

```suggestion

    environ = populate_env_with_secrets()
    environ.update({
        "DEVLANDIA_PORT": "8000",
        "TAG": tag,
        "FRUITION": core_path,
        "FILE": core_end,
    })
```

File: jbcli/jbcli/cli/jb.py [hunk 5]
```diff
@@ -563,6 +487,7 @@ def stop(ctx, clean, env):
 def yo_upgrade():
     """ Attempt to upgrade yo juicebox to the current version
     """
+    return
```
Comment (radix) [inline jbcli/jbcli/cli/jb.py:490]: was this intentional? do we just want to remove this command entirely?

File: jbcli/jbcli/tests/test_docker.py [hunk 6]
```diff
@@ -16,8 +16,8 @@ def test_up(self, check_mock):
         dockerutil.up()
         assert check_mock.mock_calls == [
             call(['docker-compose',
-                  '--project-directory', '.', '--project-name', 'stable',
-                  '-f', '../docker-compose.common.yml', '-f', 'docker-compose.yml',
+                  '--project-directory', '.', '--project-name', 'devlandia',
+                  '-f', 'docker-compose.common.yml', '-f', 'docker-compose.yml',
```
Comment (radix) [inline jbcli/jbcli/tests/test_docker.py:20]: now that we only have one docker-compose.yml, we can drop docker-compose.common.yml and just include it directly into docker-compose.yml

File: jbcli/jbcli/cli/jb.py [hunk 7]
```diff
@@ -50,6 +50,9 @@ def cli():
 def add(applications, add_desktop, runtime):
     """Checkout a juicebox app (or list of apps) and load it
     """
+    if not os.path.exists("apps"):
+        print("Please run `jb add` from the root devlandia directory.")
+        return
```
Comment (radix) [inline jbcli/jbcli/cli/jb.py:55]: instead of putting an error message here, we can just chdir to the correct place.

(this check you added would still pass when the user is in some unrelated directory that happens to have an `apps` directory, like if they're running it from their `fruition` checkout)

```suggestion
    os.chdir(os.path.join(DEVLANDIA_DIR))
```

File: jbcli/jbcli/cli/jb.py [hunk 8]
```diff
@@ -165,7 +168,9 @@ def clone(existing_app, new_app, init, track, runtime):
 def remove(applications, runtime):
     """Remove a juicebox app (or list of apps) from your local environment
     """
-
+    if not os.path.exists("apps"):
+        print("Please run `jb add` from the root devlandia directory.")
+        return
```
Comment (radix) [inline jbcli/jbcli/cli/jb.py:173]: same

```suggestion
    os.chdir(os.path.join(DEVLANDIA_DIR))
```

**Review by radix — CHANGES_REQUESTED** [review]
looking good to me. I have a few suggestions

File: .env (hunk 3 above)
Comment (obscure-shadow) [reply .env:1]: all of the previous ones had a .env file with this in it, so I included it, I'm not sure if it actually does anything, and I did not test removing it but I can

File: jbcli/jbcli/cli/jb.py (hunk 4 above)
Comment (obscure-shadow) [reply jbcli/jbcli/cli/jb.py:413]: I'll try this out, seems a bit more elegant than what I was trying. we are also relying on it for turning hstm on but we could probably also do it that way as well.

File: jbcli/jbcli/cli/jb.py (hunk 5 above)
Comment (obscure-shadow) [reply jbcli/jbcli/cli/jb.py:490]: i don't specifically remember doing anything with this but chris might have when experimenting with a few changes @chrisgemignani ?

File: .env (hunk 3 above)
Comment (radix) [reply .env:1]: actually, don't remove it. I discovered that it's necessary for stuff like `jb manage` and `jb run` to work -- or anything else that ends up invoking docker-compose, because otherwise the docker-compose.yml can't be parsed because the variables will be undefined.

Comment (radix) [reply .env:1]: OOPS sorry I meant to put this comment in response to the other thread about generating .env

I don't think we need to include this .env file inside of the devlandia repo in this PR, though, just keep auto-generating it

Comment (obscure-shadow) [reply .env:1]: aright sounds good

File: jbcli/jbcli/cli/jb.py (hunk 5 above)
Comment (chrisgemignani) [reply jbcli/jbcli/cli/jb.py:490]: Yeah; I think we want to remove this command entirely. I think I was just trying something out. Let's remove entirely @obscure-shadow

File: jbcli/jbcli/cli/jb.py [hunk 9]
```diff
@@ -546,8 +472,6 @@ def get_environment_interactively(env):
 def stop(ctx, clean, env):
     """Stop a running juicebox in this environment
     """
-    env = get_environment_interactively(env)
-    os.chdir("./environments/{}".format(env))
```
Comment (radix) [inline jbcli/jbcli/cli/jb.py:550]: why were these lines deleted? it means that `jb stop` now no longer honors its `--env` argument

Comment (radix) [reply jbcli/jbcli/cli/jb.py:550]: I guess maybe what we want is to remove the `--env` argument and just change this to `os.chdir(DEVLANDIA_DIR)`?

Comment (obscure-shadow) [reply jbcli/jbcli/cli/jb.py:550]: that makes sense, we don't really need the env exactly because there's no envs anymore and the concept of env is really just an image name used in start and we arent trying to support multiple running instances of juicebox

**Review by radix — APPROVED** [review]
This LGTM. I do want to get rid of the reliance on `.env` but I will experiment with that in another branch.

**Review by chrisgemignani — APPROVED** [review]
